test/e2e/upgradetest/framework/framework.go

```diff
@@ -135,7 +162,14 @@ func (f *Framework) UpgradeOperator() error {
 		LabelSelector: labels.SelectorFromSet(map[string]string{"name": "etcd-operator"}).String(),
 	}
 	_, err = e2eutil.WaitPodsWithImageDeleted(f.KubeCli, f.KubeNS, f.OldImage, 30*time.Second, lo)
-	return err
+	if err != nil {
+		return fmt.Errorf("failed to wait for pod with old image to get deleted: %v", err)
+	}
+	podName, err := e2eutil.WaitUntilPodReady(f.KubeCli, f.KubeNS, lo, 30*time.Second)
+	if err != nil {
```

Comment:
If pod name is not used outside but just part of the error message, why not encapsulate inside the method